src/ring_theory/graded_algebra/homogeneous_ideal.lean

```diff
@@ -0,0 +1,493 @@
+/-
+Copyright (c) 2021 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang, Eric Wieser
+-/
+
+import ring_theory.ideal.basic
+import ring_theory.ideal.operations
+import linear_algebra.finsupp
+import ring_theory.graded_algebra.basic
+
+/-!
+
+# Homogeneous ideal of a graded algebra
+
+This file defines homogeneous ideals of `graded_algebra A` where `A : ι → ideal R`and operations on
+them:
+* `mul`, `inf`, `Inf` of homogeneous ideals are homogeneous;
+* `⊤`, `⊥`, i.e. the trivial ring and `R` are homogeneous;
+* `radical` of a homogeneous ideal is homogeneous.
+-/
+
+open set_like direct_sum set
+open_locale big_operators
+
+section homogeneous_core
+
+variables {ι R : Type*} [comm_ring R]
+variables (A : ι → submodule ℤ R)
```

Comment:
Presumably you can generalize ℤ here to some arbitrary ring over which R is an algebra? Perhaps it's worth renaming R to A, A to your curly A, and then repurposing `R` to be that base ring.

Reply:
It has been generalised.